regulations/tasks.py

```diff
@@ -72,10 +74,16 @@ def submit_comment(self, body, files):
     except (ClientError, RequestException) as exc:
         logger.exception(exc)
         raise self.retry(exc=exc)
+    except Task.MaxRetriesExceededError as exc:
```

Comment:
Thinking more about it, I think you are also correct in that exceptions thrown inside an exception handler would not be caught by _peer_ handlers.